remotes/resolver.go

```diff
@@ -65,7 +65,7 @@ type FetcherByDigest interface {
 	// FetcherByDigest usually returns an incomplete descriptor.
 	// Typically, the media type is always set to "application/octet-stream",
 	// and the annotations are unset.
-	FetchByDigest(ctx context.Context, dgst digest.Digest) (io.ReadCloser, ocispec.Descriptor, error)
+	FetchByDigest(ctx context.Context, desc digest.Digest, mediatype string) (io.ReadCloser, ocispec.Descriptor, error)
```

Comment:
```suggestion
	FetchByDigest(ctx context.Context, dgst digest.Digest, opts... FetchByDigestOpt) (io.ReadCloser, ocispec.Descriptor, error)
```

Reply:
thx! fixed the typo! originally I was trying to change digest in argument to descriptor, after second thought decided to keep mediatype only but forgot to change the name back loll 
For the opts I think we need to introduce content_opts.go for client which currently is not present..I will start working on it